Bug: shutdown panic

Running the `mobile-ethereum` branch with the latest `ethereum-master` merged in (`7221fd`) in normal desktop mode.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4fa40c8]

goroutine 19324 [running]:
github.com/OpenBazaar/openbazaar-go/net/service.(*OpenBazaarService).handleNewMessage(0xc0005fb5e0, 0x5a62680, 0xc003508060)
	/Users/drwasho/Work/go/src/github.com/OpenBazaar/openbazaar-go/net/service/service.go:116 +0x10c8
created by github.com/OpenBazaar/openbazaar-go/net/service.(*OpenBazaarService).HandleNewStream
	/Users/drwasho/Work/go/src/github.com/OpenBazaar/openbazaar-go/net/service/service.go:75 +0x53
exit status 2
```

--- a/net/service/service.go
+++ b/net/service/service.go
@@ -108,12 +108,11 @@ func (service *OpenBazaarService) handleNewMessage(s inet.Stream) {
 		if err := r.ReadMsg(pmes); err != nil {
 			err = s.Reset()
 			if err != nil {
-				log.Error(err)
-			}
-			if err == io.EOF {
-				log.Debugf("Disconnected from peer %s", mPeer.Pretty())
-			} else {
-				log.Errorf("Error when reading message from %s: %s", mPeer.Pretty(), err.Error())
+				if err == io.EOF {
+					log.Debugf("Disconnected from peer %s", mPeer.Pretty())
+				} else {
+					log.Errorf("Error when reading message from %s: %s", mPeer.Pretty(), err.Error())
+				}
 			}
 			return
 		}

--- a/qa/testdata/digital.json
+++ b/qa/testdata/digital.json
@@ -5,12 +5,6 @@
 		"contractType": "DIGITAL_GOOD",
 		"format": "FIXED_PRICE",
 		"expiry": "2030-08-17T04:52:19.000Z",
-		"pricingCurrency": {
-			"code": "TBTC",
-			"divisibility": 8,
-			"name": "A",
-			"currencyType": "A"
-		},
 		"acceptedCurrencies": ["TBTC", "TBCH", "TLTC", "TZEC"]
 	},
 	"item": {
@@ -20,9 +14,7 @@
     "bigPrice": "12000000",
     "priceCurrency": {
       "code": "TBTC",
-      "divisibility": 8,
-      "name": "A",
-      "currencyType": "A"
+      "divisibility": 8
     },
 		"tags": [
 			"documentary"
@@ -41,8 +33,7 @@
 			"movies"
 		],
 		"condition": "New",
-		"skus": [
-		]
+		"skus": []
 	},
 	"taxes": [
 		{

